### ISSUE TITLE
Quality Tests are stil very buggy

### ISSUE THREAD

**Issue by jwbensley** [jwbensley/Etherate#19]
This example test stalls indefinitely:

```
bensley@ubuntu-laptop:~/C/github/etherate/Etherate$ sudo ./etherate -I 1 -r
Running in RX mode
Using device lo with address 00:00:00:00:00:00, interface index 1

Physical interface MTU unknown, test might exceed physical MTU!

Entering promiscuous mode
Source MAC 00:00:5e:00:00:02
Destination MAC 00:00:5e:00:00:01
Sending gratuitous broadcasts...
Done.

Waiting for settings from TX host
Link quality tests enabled
Settings have been synchronised

Frame size if 1514 bytes
Calculating delay between hosts...
Tx to Rx delay calculated as 0.000000004 seconds

Starting test on Sun May 14 14:25:43 2017

No.	Echo Interval
1:	0.0s
2:	0.499983196s
```

```
bensley@ubuntu-laptop:~/C/github/etherate/Etherate$ sudo ./etherate -I 1 -Q 500 500
[sudo] password for bensley: 
Running in TX mode
Using device lo with address 00:00:00:00:00:00, interface index 1

Physical interface MTU unknown, test might exceed physical MTU!

Entering promiscuous mode
Source MAC 00:00:5e:00:00:01
Destination MAC 00:00:5e:00:00:02
Sending gratuitous broadcasts...
Done.

Synchronising settings with RX host
Link quality tests enabled
Settings have been synchronised

Frame size if 1514 bytes
Calculating delay between hosts...
Tx to Rx delay calculated as 0.000000004 seconds

Starting test on Sun May 14 14:25:43 2017

No.	RTT		Jitter
1:	0.000027326s	0.000027326s
2:	0.000009499s	0.000017827s
```

**Comment by jwbensley** [comment]
That is on version 0.12.beta 2017-05

**Comment by jwbensley** [comment]
Fixed in beta 0.13, converted the timespec _sec and _nsec values into a double, the _nsec value was the problem as it loops round but _sec is linearly incrementing (of course!).
